process/block/alteredaccounts/alteredAccountsProvider_test.go

```diff
@@ -0,0 +1,430 @@
+						},
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	encodedAddr := hex.EncodeToString([]byte("addr"))
+	require.Equal(t, expectedToken.Value.String(), res[encodedAddr].Tokens[0].Balance)
+}
+
+func testExtractAlteredAccountsFromPoolShouldIncludeNFT(t *testing.T) {
+	t.Parallel()
+
+	expectedToken := esdt.ESDigitalToken{
+		Value: big.NewInt(37),
+		TokenMetaData: &esdt.MetaData{
+			Nonce: 38,
+		},
+	}
+	args := getMockArgs()
+	args.AccountsDB = &state.AccountsStub{
+		LoadAccountCalled: func(_ []byte) (vmcommon.AccountHandler, error) {
+			return &state.UserAccountStub{
+				RetrieveValueFromDataTrieTrackerCalled: func(_ []byte) ([]byte, error) {
+					tokenBytes, _ := args.Marshalizer.Marshal(expectedToken)
+					return tokenBytes, nil
+				},
+			}, nil
+		},
+	}
+	aap, _ := NewAlteredAccountsProvider(args)
+
+	res, err := aap.ExtractAlteredAccountsFromPool(&indexer.Pool{
+		Logs: []*data.LogData{
+			{
+				LogHandler: &transaction.Log{
+					Address: []byte("addr"),
+					Events: []*transaction.Event {
+						{
+							Address: []byte("addr"),
+							Identifier: []byte(core.BuiltInFunctionESDTNFTTransfer),
+							Topics: [][]byte{
+								[]byte("token0"),
+								big.NewInt(38).Bytes(),
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	encodedAddr := hex.EncodeToString([]byte("addr"))
+	require.Equal(t, expectedToken.Value.String(), res[encodedAddr].Tokens[0].Balance)
+}
+
+func testExtractAlteredAccountsFromPoolAddressHasBalanceChangeEsdtAndfNft(t *testing.T) {
+	t.Parallel()
+
+	expectedToken := esdt.ESDigitalToken{
+		Value: big.NewInt(37),
+		TokenMetaData: &esdt.MetaData{
+			Nonce: 38,
+		},
+	}
+	args := getMockArgs()
+	args.AccountsDB = &state.AccountsStub{
+		LoadAccountCalled: func(_ []byte) (vmcommon.AccountHandler, error) {
+			return &state.UserAccountStub{
+				RetrieveValueFromDataTrieTrackerCalled: func(_ []byte) ([]byte, error) {
+					tokenBytes, _ := args.Marshalizer.Marshal(expectedToken)
+					return tokenBytes, nil
+				},
+			}, nil
+		},
+	}
+	aap, _ := NewAlteredAccountsProvider(args)
+
+	res, err := aap.ExtractAlteredAccountsFromPool(&indexer.Pool{
+		Txs: map[string]data.TransactionHandler{
+			"hash0": &transaction.Transaction{
+				SndAddr: []byte("addr"),
+			},
+		},
+		Logs: []*data.LogData{
+			{
+				LogHandler: &transaction.Log{
+					Address: []byte("addr"),
+					Events: []*transaction.Event {
+						{
+							Address: []byte("addr"),
+							Identifier: []byte(core.BuiltInFunctionESDTTransfer),
+							Topics: [][]byte{
+								[]byte("esdt"),
+							},
+						},
+						{
+							Address: []byte("addr"),
+							Identifier: []byte(core.BuiltInFunctionESDTNFTTransfer),
+							Topics: [][]byte{
+								[]byte("nft"),
+								big.NewInt(38).Bytes(),
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	encodedAddr := hex.EncodeToString([]byte("addr"))
+	require.Equal(t, 2, len(res[encodedAddr].Tokens))
```

Comment:
Please check for actual token values instead of only `len`.
Also, perhaps:
```suggestion
	require.Len(t, res[encodedAddr].Tokens, 2)
```